### PR TITLE
dynamic properties test

### PR DESCRIPTION
**Pull request by heavycrystal**
also ends up testing initial load

### CODE REVIEW

**Comment by heavycrystal** [comment]
test doesn't pass in a container, tried with Docker.
works locally

File: flow/e2e/postgres/peer_flow_pg_test.go [hunk 1]
```diff
@@ -1103,180 +1105,196 @@ func (s PeerFlowE2ETestSuitePG) Test_Supported_Mixed_Case_Table() {
...
+		require.EqualValues(s.t, 2, len(workflowState.TableNameSchemaMapping))
+		// 3 from first insert of 18 rows in 1 table
+		// 1 from pre-pause
+		// 3 from second insert of 18 rows in 2 tables, batch size updated
+		require.GreaterOrEqual(s.t, len(workflowState.SyncFlowStatuses), 3+1+3)
+
+		env.CancelWorkflow()
+	}()
+
+	env.ExecuteWorkflow(peerflow.CDCFlowWorkflowWithConfig, config, nil)
```
Comment (serprex) [inline flow/e2e/postgres/peer_flow_pg_test.go:1299]:
```suggestion
	env.ExecuteWorkflow(peerflow.CDCFlowWorkflowWithConfig, config, nil)
	e2e.RequireEnvCanceled(s.t, env)
```

File: flow/e2e/postgres/peer_flow_pg_test.go [hunk 2]
```diff
@@ -1103,180 +1105,196 @@ func (s PeerFlowE2ETestSuitePG) Test_Supported_Mixed_Case_Table() {
...
+		workflowState = getWorkFlowState()
+		require.EqualValues(s.t, 14, workflowState.SyncFlowOptions.IdleTimeoutSeconds)
+		require.EqualValues(s.t, 12, workflowState.SyncFlowOptions.BatchSize)
+		require.EqualValues(s.t, 2, len(workflowState.TableMappings))
+		require.EqualValues(s.t, 2, len(workflowState.SrcTableIdNameMapping))
+		require.EqualValues(s.t, 2, len(workflowState.TableNameSchemaMapping))
+		// 3 from first insert of 18 rows in 1 table
+		// 1 from pre-pause
+		// 3 from second insert of 18 rows in 2 tables, batch size updated
+		require.GreaterOrEqual(s.t, len(workflowState.SyncFlowStatuses), 3+1+3)
```
Comment (serprex) [inline flow/e2e/postgres/peer_flow_pg_test.go:1294]: Cannot call `require` functions in goroutines; instead compare & then use `s.t.Fail` or use `e2e.Env...` helper functions

File: flow/e2e/postgres/peer_flow_pg_test.go [hunk 3]
```diff
@@ -1103,180 +1105,196 @@ func (s PeerFlowE2ETestSuitePG) Test_Supported_Mixed_Case_Table() {
...
+		e2e.EnvWaitFor(s.t, env, 1*time.Minute, "sent updates signal", func() bool {
+			return sentUpdate
+		})
+
+		// add rows to both tables before resuming - should handle
+		addRows(18)
+
+		e2e.EnvWaitFor(s.t, env, 1*time.Minute, "resumed workflow", func() bool {
+			flowStatus := getFlowStatus()
+			return flowStatus == protos.FlowStatus_STATUS_RUNNING
```
Comment (serprex) [inline flow/e2e/postgres/peer_flow_pg_test.go:1276]:
```suggestion
			return getFlowStatus() == protos.FlowStatus_STATUS_RUNNING
```

File: flow/e2e/postgres/peer_flow_pg_test.go [hunk 4]
```diff
@@ -1103,180 +1106,196 @@ func (s PeerFlowE2ETestSuitePG) Test_Supported_Mixed_Case_Table() {
...
+
+		workflowState := getWorkFlowState()
+		assert.EqualValues(s.t, 7, workflowState.SyncFlowOptions.IdleTimeoutSeconds)
+		assert.EqualValues(s.t, 6, workflowState.SyncFlowOptions.BatchSize)
+		assert.EqualValues(s.t, 1, len(workflowState.TableMappings))
+		assert.EqualValues(s.t, 1, len(workflowState.SrcTableIdNameMapping))
+		assert.EqualValues(s.t, 1, len(workflowState.TableNameSchemaMapping))
+		// we have limited batch size to 6, so atleast 3 syncs needed
+		assert.GreaterOrEqual(s.t, len(workflowState.SyncFlowStatuses), 3)
+
```
Comment (serprex) [inline flow/e2e/postgres/peer_flow_pg_test.go:1252]: Can check `s.t.Failed()` here to avoid entering `EnvWaitFor` in a failed test

**Review by serprex — APPROVED** [review]

**Pull request merged**